test/governance/utils/Votes.behavior.js

```diff
@@ -6,62 +6,49 @@ const { fromRpcSig } = require('ethereumjs-util');
 const ethSigUtil = require('eth-sig-util');
 const Wallet = require('ethereumjs-wallet').default;
 
-const { EIP712Domain, domainSeparator } = require('../../helpers/eip712');
+const { getDomain, domainType, domainSeparator } = require('../../helpers/eip712');
 
 const Delegation = [
   { name: 'delegatee', type: 'address' },
   { name: 'nonce', type: 'uint256' },
   { name: 'expiry', type: 'uint256' },
 ];
 
-const version = '1';
-
 function shouldBehaveLikeVotes() {
   describe('run votes workflow', function () {
     it('initial nonce is 0', async function () {
       expect(await this.votes.nonces(this.account1)).to.be.bignumber.equal('0');
     });
 
     it('domain separator', async function () {
-      expect(await this.votes.DOMAIN_SEPARATOR()).to.equal(
-        await domainSeparator({
-          name: this.name,
-          version,
-          chainId: this.chainId,
-          verifyingContract: this.votes.address,
-        }),
-      );
+      expect(await this.votes.DOMAIN_SEPARATOR()).to.equal(await getDomain(this.votes).then(domainSeparator));
```

Comment:
Why use `.then` here? Do you find it more readable in this case? I want to understand.

```suggestion
      expect(await this.votes.DOMAIN_SEPARATOR()).to.equal(domainSeparator(await getDomain(this.votes)));
```

Reply:
Yes, I find it more readeable to have the await first then a list of chained operations.

I think it mostly comes down to the fact that using "then" makes the successive operations flow from left to right (in the direction i'm reading) ... It feels way more natural then having to keep a pile of operation in my head before getting to the first part.

Reply:
But that's a general problem with the notation `f(g(x))`. It seems to me like we just have to come to terms with the fact that operations are right to left.